File: abc096/c.py
import sys

h, w = map(int, input().split())

s = []
for i in range(h):
    s.append(input())

ans = True
flag = False

for i in range(h):
    for j in range(w):
        if s[i][j] == '#':
            search_list = []
            if i==0:
                if j==0:
                    search_list = [s[i+1][j], s[i][j+1]]
                elif j==w-1:
                    search_list = [s[i+1][j], s[i][j-1]]
                else:
                    search_list = [s[i][j-1], s[i+1][j], s[i][j+1]]

            if i==h-1 and j==0: search_list = [s[i-1][j], s[i][j+1]]
            if i==w-1 and j==w-1: search_list = [s[i-1][j], s[i][j-1]]
            if j==0: search_list = [s[i-1][j], s[i+1][j], s[i][j+1]]
            if i==h-1: search_list = [s[i-1][j], s[i][j-1], s[i][j+1]]
            if j==w-1: search_list = [s[i-1][j], s[i][j-1], s[i+1][j]]

    if not(ans):
        break

if ans:
    print('Yes')
else:
    print('No')

File: abc096/c.cpp
#include <iostream>

using namespace std;

int main() {
  int h, w;
  cin >> h >> w;

  string s[50];

  bool ans = true;
  bool flag = false;

  for(int i=0; i < h; i++){
    for (int j=0; j < w; j++){
      if (s[i][j] == '#'){

        if ((i==0) && (j==0)) {
          if((s[i+1][j] == '.') && s[i][j+1] == '.'){
            ans = false;
            break;
          }
        }

        if (i == 0){
          if(s[i][j-1] == '.' && s[i+1][j] == '.' && s[i][j+1] == '.'){
            ans = false;
            break;
          }
          break;
        }

        if (j == 0){
          if(s[i-1][j] == '.' && s[i+1][j] == '.' && s[i][j+1] == '.'){
            ans = false;
            break;
          }
          break;
        }

        if (i == h-1){
          if(s[i][j-1] == '.' && s[i-1][j] == '.' && s[i][j+1] == '.'){
            ans = false;
            break;
          }
          break;
        }

        if (j == w-1){
          if(s[i][j-1] == '.' && s[i+1][j] == '.' && s[i-1][j] == '.'){
            ans = false;
            break;
          }
          break;
        }

        if (j == w-1){
          if(s[i][j-1] == '.' && s[i+1][j] == '.' && s[i-1][j] == '.'){
            ans = false;
            break;
          }
          break;
        }

        if(s[i-1][j] && s[i][j-1] == '.' &&  s[i+1][j] == '.' && s[i-1][j] == '.'){
          ans = false;
          break;
        }
       }

      }
    if (!(ans)){
        break;
      }
  }

  if(ans){
    cout << "Yes" << endl;
  } else {
    cout << "No" << endl;
  }

}
